python/ml_croissant/ml_croissant/_src/datasets_test.py

```diff
@@ -10,84 +10,31 @@
 
 
 # End-to-end tests on real data. The data is in `tests/graphs/*/metadata.json`.
-@pytest.mark.parametrize(
-    ["filename", "error"],
-    [
-        # Metadata.
-        [
-            "metadata_missing_property_name.json",
-            'Property "https://schema.org/name" is mandatory, but does not exist.',
-        ],
-        [
-            "metadata_bad_type.json",
-            "No metadata is defined in the dataset",
-        ],
-        # Distribution.
-        [
-            "distribution_missing_property_content_url.json",
-            (
-                'Property "https://schema.org/contentUrl" is mandatory, but does not '
-                "exist."
-            ),
-        ],
-        [
-            "distribution_bad_type.json",
-            'Node should have an attribute `"@type" in',
-        ],
-        [
-            # When the name misses, the context should still appear without the name.
-            "distribution_missing_name.json",
-            (
-                r"\[dataset\(mydataset\) \> distribution\(\)\] Property "
-                r'"https://schema.org/name" is mandatory'
-            ),
-        ],
-        # Record set.
-        [
-            "recordset_missing_property_name.json",
-            'Property "https://schema.org/name" is mandatory, but does not exist.',
-        ],
-        [
-            "recordset_bad_type.json",
-            'Node should have an attribute `"@type" in',
-        ],
-        [
-            "recordset_missing_context_for_datatype.json",
-            "The field does not specify any http://mlcommons.org/schema/dataType",
-        ],
-        # ML field.
-        [
-            "mlfield_missing_property_name.json",
-            'Property "https://schema.org/name" is mandatory, but does not exist.',
-        ],
-        [
-            "mlfield_bad_type.json",
-            'Node should have an attribute `"@type" in',
-        ],
-        [
-            "mlfield_missing_source.json",
-            'Node "a-record-set/first-field" is a field and has no source.',
-        ],
-        [
-            "mlfield_bad_source.json",
-            "Malformed source data: THISDOESNOTEXIST.",
-        ],
-    ],
-)
-def test_static_analysis_old(filename, error):
-    base_path = epath.Path(__file__).parent / "tests/graphs"
-    with pytest.raises(ValidationError, match=error):
-        datasets.Dataset(base_path / filename)
-
-
 def get_error_msg(folder):
     with open(f"{folder}/output.txt", "r") as file:
         return file.read().strip()
 
 
-# TODO(https://github.com/mlcommons/croissant/issues/14): Progressively move tests from
-# test_static_analysis_old to test_static_analysis
-@pytest.mark.parametrize("folder", ["distribution_bad_contained_in"])
+@pytest.mark.parametrize("folder", [
+                          # Distribution.
```

Comment:
Should you locate every folder with a regular expression, so that adding a new folder is transparent?

Reply:
Hi! Not sure what you mean with locating each folder w. a regex?
Another option might be to just iterate over tests/graphs... but I like how we are writing down all tests in the current version, grouping similar tests together and adding comments, it makes it easier to read I think?